### PR TITLE
archive parsing: use strings.Trim...Func() instead of strings.Trim()

### PR DESCRIPTION
**Pull request by nalind**
When trimming string values as part of parsing them, use the `string.Trim()` variants that take a callback function instead of a string of characters to trim, so that they won't have to internally allocate a new helper structure on the heap every time we call them.

### CODE REVIEW

**Comment by mtrmac** [comment]
Will this be necessary/useful in the future? https://github.com/golang/go/issues/46446 should make at least the single-character case more efficient using the simple code.

https://github.com/golang/go/commit/4002616f9a34410797e7bbff142c374a1de3ac6b also optimizes the multiple-ASCII-character case (although in that case it’s not _obvious_ without measurement that the ASCII table approach is more efficient than `Trim…Func` for the very small strings used here).

File: archive/tar/strconv.go [hunk 1]
```diff
@@ -216,7 +216,7 @@ func parsePAXTime(s string) (time.Time, error) {
 	}
 
 	// Parse the nanoseconds.
-	if strings.Trim(sn, "0123456789") != "" {
+	if strings.TrimFunc(sn, func(r rune) bool { return r >= 0x30 && r <= 0x39 }) != "" {
```
Comment (mtrmac) [inline archive/tar/strconv.go:219]: This could be `IndexFunc(…) != -1`, eliminating one more allocation — or, well, just a loop.

Comment (nalind) [reply archive/tar/strconv.go:219]: Well, a secondary goal was to keep the patch minimal to help keep the next rebase simple, but you're right.  I'm not wedded to the specific change so much as the result, so if you want that change, I'm okay with it.

**Comment by nalind** [comment]
> Will this be necessary/useful in the future? [golang/go#46446](https://github.com/golang/go/issues/46446) should make at least the single-character case more efficient using the simple code.

I would like for it to not be necessary in the future, of course.

File: archive/tar/strconv.go (hunk 1 above)
Comment (mtrmac) [reply archive/tar/strconv.go:219]: Equally, there would be some benefit to minimizing the difference from original archive/tar …

It’s not at all up to me, I know nothing about the history of tar-split and almost nothing about its goals and philosophy.

**Comment by mtrmac** [comment]
Looking now, neither of the changes is in the released Go 1.17.  Go 1.18 seems to be scheduled for February.

**Comment by vbatts** [comment]
so, we could just wait until go1.18, and this improvement will be transparent and maintained by someone else?

**Comment by vbatts** [comment]
@nalind I realize I sat on this contribution :confused:  
I value the contribution. :orange_heart: 
Now that we're up to go1.20, is this still needed?

**Pull request closed without merging**